### PR TITLE
Switch mode after autonomous disarm

### PR DESCRIPTION
**Pull request by MaEtUgR**
## Describe problem solved by this pull request
1. There can be seemingly random switches to Hold mode when the vehicle has an RC loss while sitting on the ground, disarmed, in Manual mode.
2. After an RTL the vehicle stays in Return mode even though it's already done and it's not possible to take off again in that mode. Users get annoyed by this and there are corner cases with using the takeoff button in the ground station.
3. There are still MAVSDK use cases which will fail arming even after arming in Land mode was temporarily allowed again in https://github.com/PX4/PX4-Autopilot/pull/19449

## Describe your solution
1. Only automatic switch to Hold mode for a connected tablet and no stick input on initialization and not later on when having RC loss.
2. After landing in an autonomous mode (most of the time RTL, Land, Mission) switch back to Hold mode upon disarming.
3. That should be solved with 2. and arming in Land mode can be disallowed again.

## Test data / coverage
I only did a quick SITL test. More testing to be done. Please review and help me think of corner cases. MAVSDK should hopefully not fail anymore with this.

### CODE REVIEW

**Review by julianoes — DISMISSED** [review]
This looks correct, thanks a lot @MaEtUgR.

My quick testing with MAVSDK (with and without QGC with joystick connected) worked fine.

**Review by sfuhrer — APPROVED** [review]
I don't have 100% visibility in the Commander logic (well, who has), but the changes proposed with those 4 commits make all sense to me, and especially switching back to Hold (Loiter) after landing is intuitively for the UX the better option than staying in a landing mode.

**Comment by MaEtUgR** [comment]
I had a trivial renaming problem with https://github.com/PX4/PX4-Autopilot/pull/19707 left in my rebase. Dismissing all reviews 🙈

**Review by julianoes — APPROVED** [review]

**Review by sfuhrer — APPROVED** [review]

**Pull request merged**
